src/main/scala/higherkindness/rules_scala/workers/zinc/compile/AnalysisStore.scala

```diff
@@ -690,10 +690,38 @@ class AnxAnalyses(format: AnxAnalysisStore.Format) {
     (stream, value) => format.write(writer.toSourceInfos(value), stream)
   )
 
-  def stamps = new Store[Stamps](
-    stream => reader.fromStamps(format.read(Schema.Stamps.getDefaultInstance, stream)),
-    (stream, value) => format.write(writer.toStamps(value), stream)
-  )
+  def stamps(): Store[Stamps] = {
+    new Store[Stamps](
+      stream => reader.fromStamps(format.read(Schema.Stamps.getDefaultInstance, stream)),
+      (stream, value) =>
+        format.write(
+          sortStamps(writer.toStamps(value)),
+          stream
+        )
+    )
+  }
+
+  def sortStamps(stamps: Schema.Stamps): Schema.Stamps = {
+    val sortedProductStamps = sortStampTypeMap(stamps.getProductStampsMap).asJava
+    val sortedBinaryStamps = sortStampTypeMap(stamps.getBinaryStampsMap).asJava
+    val sortedSourceStamps = sortStampTypeMap(stamps.getSourceStampsMap).asJava
+
+    Schema.Stamps
+      .newBuilder(stamps)
+      .clearProductStamps()
+      .putAllProductStamps(sortedProductStamps)
+      .clearBinaryStamps()
+      .putAllBinaryStamps(sortedBinaryStamps)
+      .clearSourceStamps()
+      .putAllSourceStamps(sortedSourceStamps)
+      .build()
+  }
+
+  def sortStampTypeMap(
+    stampTypeMap: util.Map[String, Schema.Stamps.StampType]
+  ): TreeMap[String, Schema.Stamps.StampType] = {
+    TreeMap[String, Schema.Stamps.StampType]() ++ stampTypeMap.asScala
```

Comment:
```suggestion
    TreeMap.from(stampTypeMap.asScala)
```
Though going from a Java type to a Scala type just to put it ordered map and then convert back to a Java type seems wasteful. Why not go directly to the ordered Java map?

Reply:
Good questions, I don't remember this code was written so long ago. Lemme go take another look. I bet we can go directly to the ordered Java map and that was a rough edge I forgot about last year.

Reply:
I updated this to use a `java.util.TreeMap` directly.